packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.appButton.onHoverEnter should successfully register a app button hover handler when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            let handlerCalled = false;
+            pages.appButton.onHoverEnter(() => {
+              handlerCalled = true;
+            });
+            utils.sendMessage('appButtonHoverEnter', '');
+            expect(handlerCalled).toBeTruthy();
+          });
+        } else {
+          it(`pages.appButton.onHoverEnter does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.appButton.onHoverEnter(emptyCallback)).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.appButton.onHoverLeave function', () => {
+      it('pages.appButton.onHoverLeave should not allow calls before initialization', () => {
+        expect(() => pages.appButton.onHoverLeave(emptyCallback)).toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.appButton.onHoverLeave should successfully register a app button hover leave handler when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            let handlerCalled = false;
+            pages.appButton.onHoverLeave(() => {
+              handlerCalled = true;
+            });
+            utils.sendMessage('appButtonHoverLeave', '');
+            expect(handlerCalled).toBeTruthy();
+          });
+        } else {
+          it(`pages.appButton.onHoverLeave does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.appButton.onHoverLeave(emptyCallback)).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.appButton.isSupported function', () => {
+      it('pages.appButton.isSupported should return false if the runtime says pages.appButton is not supported', () => {
+        utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
```

Comment:
I think you know what my comment is here. :)

Reply:
Added in the latest commit